tests/unit/oidc/models/test_github.py

```diff
@@ -20,19 +20,67 @@
 from warehouse.oidc.models import _core, github
 
 
+@pytest.mark.parametrize(
+    ("workflow_ref", "expected"),
+    [
+        # Well-formed workflow refs, including exceedingly obnoxious ones
+        # with `@` or extra suffixes or `git` refs that look like workflows.
+        ("foo/bar/.github/workflows/basic.yml@refs/heads/main", "basic.yml"),
+        ("foo/bar/.github/workflows/has-dash.yml@refs/heads/main", "has-dash.yml"),
+        ("foo/bar/.github/workflows/has--dashes.yml@refs/heads/main", "has--dashes.yml"),
+        ("foo/bar/.github/workflows/has--dashes-.yml@refs/heads/main", "has--dashes-.yml"),
+        ("foo/bar/.github/workflows/has.period.yml@refs/heads/main", "has.period.yml"),
+        ("foo/bar/.github/workflows/has..periods.yml@refs/heads/main", "has..periods.yml"),
+        ("foo/bar/.github/workflows/has..periods..yml@refs/heads/main", "has..periods..yml"),
+        (
+            "foo/bar/.github/workflows/has_underscore.yml@refs/heads/main",
+            "has_underscore.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/nested@evil.yml@refs/heads/main",
+            "nested@evil.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/nested.yml@evil.yml@refs/heads/main",
+            "nested.yml@evil.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/extra@nested.yml@evil.yml@refs/heads/main",
+            "extra@nested.yml@evil.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/extra.yml@nested.yml@evil.yml@refs/heads/main",
+            "extra.yml@nested.yml@evil.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/basic.yml@refs/heads/misleading@branch.yml",
+            "basic.yml",
+        ),
+        (
+            "foo/bar/.github/workflows/basic.yml@refs/heads/misleading@branch@causestwomatches.yml",
+            "basic.yml",
+        ),
+        ("foo/bar/.github/workflows/foo.yml.yml@refs/heads/main", "foo.yml.yml"),
+        (
+            "foo/bar/.github/workflows/foo.yml.foo.yml@refs/heads/main",
+            "foo.yml.foo.yml",
+        ),
+        # Malformed workflow refs.
+        ("foo/bar/.github/workflows/@refs/heads/main", None),
+        ("foo/bar/.github/workflows/nosuffix@refs/heads/main", None),
+        ("foo/bar/.github/workflows/.yml@refs/heads/main", None),
+        ("foo/bar/.github/workflows/main.yml", None),
+    ],
+)
+def test_extract_workflow_filename(workflow_ref, expected):
+    assert github._extract_workflow_filename(workflow_ref) == expected
+
+
 @pytest.mark.parametrize("claim", ["", "repo", "repo:"])
 def test_check_sub(claim):
     assert github._check_sub(pretend.stub(), claim, pretend.stub()) is False
 
 
-def test_lookup_strategies():
-    assert (
-        len(github.GitHubPublisher.__lookup_strategies__)
-        == len(github.PendingGitHubPublisher.__lookup_strategies__)
-        == 2
-    )
-
-
```

Comment:
Intentional deletion?

Reply:
Yep, this was duplicated with the class-scoped test immediately below 🙂:

https://github.com/pypi/warehouse/pull/16462/files#diff-cc2dfefbb24e9b4bb64ca653fe64240ed824fda8f40e7ef0f233c55e94ad138fR85